overlays/bootstrap.nix

```diff
@@ -474,9 +482,10 @@ in {
             } // args);
             alex = bootstrap.packages.alex-tool {};
             happy-tool = args: tool "happy" ({
-                version = "1.19.11";
+                version = "1.19.12";
```

Comment:
Why do we need 1.19.12? There is some limitation on happy to build ghc-8.6.5 due to a Parser change.

Reply:
8.8.x doesn't like <1.19.11; don't remember the exact failure, but the fix is called out in the changelog: https://hackage.haskell.org/package/happy-1.19.12/changelog

Reply:
building ghc-8.6.5 with happy-1.19.12 will eventually fail with:
```
compiler/stage1/build/Parser.hs:1473:48: error:
    Not in scope: type variable ‘a’
     |
1473 | newtype HappyWrap209 = HappyWrap209 (([Located a],Bool))
     |                                                ^
```
So I'm afraid we'll have to special case this somehow :( This is really unfortunate.